File: R/R/l_isLoonWidget.R

#' @title Check if a widget path is a valid loon widget
#'
#' @description This function can be useful to check whether a loon widget is
#'   has been closed by the user.
#'
#' @template param_widget
#'
#' @return boolean, TRUE if the argument is a valid loon widget path, FALSE
#'   otherwise
#'
#' @export
l_isLoonWidget <- function(widget) {

    if(is.null(widget)) return(FALSE)
    if(is.list(widget)) return(FALSE)

    as.logical(as.character(tcl('::loon::isKnownWidget', widget)))
}

File: Python/loon/l_isLoonWidget.py
from tk import tk
def l_isLoonWidget(widget):
    """
    Check if a widget path is a valid loon widget
    
        This function can be useful to check whether a loon widget is
        has been closed by the user.

    Args:
        widget: widget path name as a string
    Returns:
        boolean, True if the argument is a valid loon widget path, False 
        otherwise
    """
    if(isinstance(type(widget), list)):
        return False
    else:
        isLoon = tk.tk.call('::loon::isKnownWidget', widget)
        if (isLoon): return True
        else: return False
